### PR TITLE
Copy log.syslog.appname to process.name

### LINKED ISSUE #9565: System Auth integration not populating process.name
Hello,
Last week I saw (based on an slack message https://elasticstack.slack.com/archives/C02J2JBS0FP/p1712234933586649) that when creating a user in Linux (version 1.54 of system integration) the process.name was not properly populated.
The event.category and event.type is set based on the value of the process name and therefore the events were without category/type
What I have observed is that the value of the process is set in log.syslog.appname, but I'm not able to find references to this field in the pipeline.

Also, doing a pipeline simulation the field log.syslog.appname is not set... so I do not understand

image

The event original is this one
`Apr  4 15:13:14 serverxyz useradd[2586]: new user: name=testoriginal, UID=6185, GID=6185, home=/home/testoriginal, shell=/bin/bash`

What is wrong? What is the origin of the field log.syslog.appname?

### PR DESCRIPTION
**Pull request by nicholasberlin**
## Proposed commit message

Before the syslog processor was added to the default filebeat yaml configuration, process.name was set by a Grok ingest pipeline. However, when the syslog processor was added a condition was also added to the Grok ingest pipeline, `if: ctx.log?.syslog == null`. This condition always evaluates to false when the data is supplied by the syslog processor, and therefore `process.name` is not extraced from the syslog message. This commit copies the process name from `log.syslog.appname` to `process.name` before an pipeline processors which use it are executed. With this change, `process.name`, `event.category`, and `event.type` are now populated when data arrives from the syslog processor.

And add tests.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
~/git/integrations/main/packages/system$ elastic-package test pipeline -d auth -v
```

## Related issues
- Closes #9565

### CODE REVIEW

**Comment by andrewkroh** [comment]
Can you fix the .github/CODEOWNERS file while you are in there. I think the auth data stream belongs under the sec-linux-platform team instead of windows.

https://github.com/elastic/integrations/blob/aba9540f850f673f1a626248b80c962176f87e63/.github/CODEOWNERS#L331

File: packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                "eventSource": "Application",
+                "eventID":     "1011"
+              },
+              "examplePriority@32473": {
+                "class": "high"
+              }
+            }
+          }
+        },
+        "message": "<165>1 2022-01-11T22:14:15.003Z mymachine.example.com eventslog 1024 ID47 [exampleSDID@32473 iut=\"3\" eventSource=\"Application\" eventID=\"1011\"][examplePriority@32473 class=\"high\"] this is the message"
```
Comment (andrewkroh) [inline packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json:33]: I think I recall that the input configuration was set to syslog decode the `event.original` field. If so, then are more accurate representation of the data leaving the agent would be

```
{
  "@timestamp": "2022-01-11T22:14:15.003Z",
  "event": {
    "original": "<165>1 2022-01-11T22:14:15.003Z mymachine.example.com eventslog 1024 ID47 [exampleSDID@32473 iut=\"3\" eventSource=\"Application\" eventID=\"1011\"][examplePriority@32473 class=\"high\"] this is the message"
  },
  "log": {
    "syslog": {
      "appname": "eventslog",
      "facility": {
        "code": 20,
        "name": "local4"
      },
      "hostname": "mymachine.example.com",
      "msgid": "ID47",
      "priority": 165,
      "procid": "1024",
      "severity": {
        "code": 5,
        "name": "Notice"
      },
      "structured_data": {
        "examplePriority@32473": {
          "class": "high"
        },
        "exampleSDID@32473": {
          "eventID": "1011",
          "eventSource": "Application",
          "iut": "3"
        }
      },
      "version": "1"
    }
  },
  "message": "this is the message"
}
```

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -303,6 +303,12 @@ processors:
           ctx.event.action = "ssh_login";
           ctx.event.outcome = "failure";
         }
+  - set:
+      tag: set_process_name-log_syslog_appname
+      field: process.name
+      copy_from: log.syslog.appname
+      override : false
```
Comment (andrewkroh) [inline packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml:310]:
```suggestion
      override: false
```

Comment (nicholasberlin) [reply packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml]: ecbef38bd7fa0ba9711b848e17d72945887a0009

File: packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json (hunk 1 above)
Comment (nicholasberlin) [reply packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json]: Done. ecbef38bd7fa0ba9711b848e17d72945887a0009

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**